OpenCL/m19500_a3-pure.cl

```diff
@@ -0,0 +1,344 @@
+  }
+
+  barrier (CLK_LOCAL_MEM_FENCE);
+
+  if (gid >= gid_max) return;
+
+  /**
+   * base
+   */
+
+  const u32 pw_len = pws[gid].pw_len;
+
+  u32x w[64] = { 0 };
+
+  for (int i = 0, idx = 0; i < pw_len; i += 4, idx += 1)
+  {
+    w[idx] = pws[gid].i[idx];
+  }
+
+  const int salt_len = esalt_bufs[digests_offset].salt_len;
+
+  const int site_key_len = esalt_bufs[digests_offset].site_key_len;
+
+  u32 s[64] = { 0 };
+  u32 k[64] = { 0 };
+
+  const u32 glue[16] = {0x2d2d0000};
+
+  for (int i = 0, idx = 0; i < salt_len; i += 4, idx += 1)
+  {
+    s[idx] = swap32_S (esalt_bufs[salt_pos].salt_buf[idx]);
+  }
+
+  for (int i = 0, idx = 0; i < site_key_len; i += 4, idx += 1)
+  {
+    k[idx] = swap32_S (esalt_bufs[salt_pos].site_key_buf[idx]);
+  }
+
+  /**
+   * loop
+   */
+
+  u32x w0l = w[0];
+
+  for (u32 il_pos = 0; il_pos < il_cnt; il_pos += VECT_SIZE)
+  {
+    const u32x w0r = words_buf_r[il_pos / VECT_SIZE];
+
+    const u32x w0 = w0l | w0r;
+
+    w[0] = w0;
+
+    sha1_ctx_t ctx0;
+    sha1_ctx_t ctx;
+
+    sha1_init (&ctx0);
+
+    sha1_update_vector (&ctx0, k, site_key_len);
+    sha1_update_vector (&ctx0, glue, 2);
+    sha1_update_vector (&ctx0, s, salt_len);
+    sha1_update_vector (&ctx0, glue, 2);
+    sha1_update_vector (&ctx0, w, pw_len);
+    sha1_update_vector (&ctx0, glue, 2);
+    sha1_update_vector (&ctx0, k, site_key_len);
+
+    sha1_final (&ctx0);
+
+    for (u32 iter = 0; iter < 9; iter++)
+    {
+      const u32 a = ctx0.h[0];
+      const u32 b = ctx0.h[1];
+      const u32 c = ctx0.h[2];
+      const u32 d = ctx0.h[3];
+      const u32 e = ctx0.h[4];
+
+      sha1_init (&ctx);
+
+      ctx.w0[0] = uint_to_hex_lower8_le ((a >> 16) & 255) <<  0
+                | uint_to_hex_lower8_le ((a >> 24) & 255) << 16;
+      ctx.w0[1] = uint_to_hex_lower8_le ((a >>  0) & 255) <<  0
+                | uint_to_hex_lower8_le ((a >>  8) & 255) << 16;
+      ctx.w0[2] = uint_to_hex_lower8_le ((b >> 16) & 255) <<  0
+                | uint_to_hex_lower8_le ((b >> 24) & 255) << 16;
+      ctx.w0[3] = uint_to_hex_lower8_le ((b >>  0) & 255) <<  0
+                | uint_to_hex_lower8_le ((b >>  8) & 255) << 16;
+      ctx.w1[0] = uint_to_hex_lower8_le ((c >> 16) & 255) <<  0
+                | uint_to_hex_lower8_le ((c >> 24) & 255) << 16;
+      ctx.w1[1] = uint_to_hex_lower8_le ((c >>  0) & 255) <<  0
+                | uint_to_hex_lower8_le ((c >>  8) & 255) << 16;
+      ctx.w1[2] = uint_to_hex_lower8_le ((d >> 16) & 255) <<  0
+                | uint_to_hex_lower8_le ((d >> 24) & 255) << 16;
+      ctx.w1[3] = uint_to_hex_lower8_le ((d >>  0) & 255) <<  0
+                | uint_to_hex_lower8_le ((d >>  8) & 255) << 16;
+      ctx.w2[0] = uint_to_hex_lower8_le ((e >> 16) & 255) <<  0
+                | uint_to_hex_lower8_le ((e >> 24) & 255) << 16;
+      ctx.w2[1] = uint_to_hex_lower8_le ((e >>  0) & 255) <<  0
+                | uint_to_hex_lower8_le ((e >>  8) & 255) << 16;
+
+      ctx.len = 40;
```

Comment:
At this position you can hardcore the glue, it will always be at exactly this point.

```
      ctx.w2[2] = glue[0];

      ctx.len = 40 + 2;
```